### PR TITLE
Fix use of ortools binary release.

### PR DESCRIPTION
**Pull request by aosmw**
Use FetchContent_Populate method to allow manipulation of CMAKE_PREFIX_PATH in order for find_package(ortools) to find its packaged dependencies such as absl and protobuf

### CODE REVIEW

**Comment by aosmw** [comment]
I like the idea of NOT having to build ortools.

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ option(BUILD_TESTS "Build tests" ON)
 option(BUILD_PYTHON "Build Python SWIG module" OFF)
 option(BUILD_DOC "Build Documentation" OFF)
 option(BUILD_SHARED_LIBS "Build shared library(.so)" ON)
-option(USE_ORTOOLS_RELEASE_IF_NOT_FOUND "If or-tools is not found, use the release" OFF)
+option(USE_ORTOOLS_RELEASE_IF_NOT_FOUND "If or-tools is not found, use the release" ON)
```
Comment (aosmw) [inline CMakeLists.txt:28]: Not sure if you want this ON by default or not.  Feel free to revert.

**Comment by Gonzalo-Mier** [comment]
I was trying to make the ROS compilers to be able to compile the project automatically. Unfortunately, it still gives errors. The actual one is:

```
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "ZLIB" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "base" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "core_headers"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "absl_check" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "absl_log" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "check" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "die_if_null"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "flags" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target
17:29:05   "flags_commandlineflag" that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target
17:29:05   "flags_marshalling" that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "flags_parse"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target
17:29:05   "flags_reflection" that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "flags_usage"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "log" that is not
17:29:05   in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "log_flags" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "log_globals"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "log_initialize"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target
17:29:05   "log_internal_message" that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "cord" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "random_random"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "raw_hash_set"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "hash" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "leak_check" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "memory" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "meta" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "stacktrace" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "status" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "statusor" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "str_format" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "strings" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "synchronization"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "time" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "any" that is not
17:29:05   in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "libprotobuf"
17:29:05   that is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "re2" that is not
17:29:05   in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "CbcSolver" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "OsiCbc" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "ClpSolver" that
17:29:05   is not in any export set.
17:29:05 
17:29:05 
17:29:05 CMake Error in CMakeLists.txt:
17:29:05   export called with target "ortools" which requires target "OsiClp" that is
17:29:05   not in any export set.
17:29:05 
17:29:05 
17:29:05 -- Generating done
17:29:05 CMake Generate step failed.  Build files cannot be regenerated correctly.
17:29:05 ---
```

Ideally, we would import it from `ortools_vendor`, but meanwhile we make that working, we need another solution for f2c.

Still, thanks for your help!

**Comment by aosmw** [comment]
Do you have a link that I can click on for the above output that relates to the "ROS compilers"? Is that the bloom mechanism?

**Comment by Gonzalo-Mier** [comment]
[https://build.ros2.org/job/Hdev__fields2cover__ubuntu_jammy_amd64/35/console](https://build.ros2.org/job/Hdev__fields2cover__ubuntu_jammy_amd64/35/console)

**Comment by aosmw** [comment]
Ok.  I think I understand what is happening.

With my commit that uses the released tarball of ortools I ran tests, "successfully", in the build directory and those know how to find the ortools.  When fields2cover installs its not currently installing the libortools.so from the release tarball into a place that it can be found at runtime.

Installations of libabsl-dev and protobuf-dev may not be needed or compatible with the ones that are in the release tarball. https://github.com/Fields2Cover/Fields2Cover/commit/fe3c2ca48bd10c6d75410b2c8613a7cd7ba4c7d7

Here you can see the difference in linkage of the library in the build directory vs once it is "installed"
```
:~/dev/contrib_github$ ldd build/fields2cover/lib/libFields2Cover.so | grep ortools
        libortools.so.9 => /home/wakem/dev/contrib_github/build/fields2cover/_deps/ortools-src/lib/libortools.so.9 (0x00007f5827184000)
~/dev/contrib_github$ ldd install/fields2cover/lib/libFields2Cover.so | grep ortools
        libortools.so.9 => not found
```

The released tarball of ortools appears to provide static libraries for things like libabsl and libproto
```
:~/dev/contrib_github$ ls build/fields2cover/_deps/ortools-src/lib/libabsl*.a | wc -l
86
```
~/dev/contrib_github$ ls build/fields2cover/_deps/ortools-src/lib/libproto*.a |wc -l
3

The solvers are also built as static libraries.  

I predict that all of those are rolled into the shared library libortools
```
~/dev/contrib_github$ ls -al build/fields2cover/_deps/ortools-src/lib/*.so.*
lrwxrwxrwx 1 wakem wakem       22 Mar 11 19:49 build/fields2cover/_deps/ortools-src/lib/libortools.so.9 -> libortools.so.9.9.3963
-rw-r--r-- 1 wakem wakem 45402608 Mar 11 19:49 build/fields2cover/_deps/ortools-src/lib/libortools.so.9.9.3963
lrwxrwxrwx 1 wakem wakem       31 Mar 11 19:49 build/fields2cover/_deps/ortools-src/lib/libortools_flatzinc.so.9 -> libortools_flatzinc.so.9.9.3963
-rw-r--r-- 1 wakem wakem  3031568 Mar 11 19:49 build/fields2cover/_deps/ortools-src/lib/libortools_flatzinc.so.9.9.3963
```

I'll have a play around with installing the ortools stuff to see if I can get it working.

**Comment by Gonzalo-Mier** [comment]
Whenever you want to try, feel free to commit on the branch and I can release it again.

**Pull request merged**